src/apps/answers/crud/answers.py

```diff
@@ -490,10 +490,17 @@ async def get_submitted_activity_with_last_date(
     async def get_submitted_flows_with_last_date(
         self, applet_id: uuid.UUID, target_subject_id: uuid.UUID | None
     ) -> list[tuple[str, datetime.datetime]]:
-        query: Query = select(AnswerSchema.flow_history_id, func.max(AnswerSchema.created_at))
-        query = query.where(AnswerSchema.applet_id == applet_id, AnswerSchema.flow_history_id.isnot(None))
+        subquery: Query = select(AnswerSchema.submit_id)
+        subquery = subquery.where(
+            AnswerSchema.applet_id == applet_id,
+            AnswerSchema.flow_history_id.isnot(None),
+            AnswerSchema.is_flow_completed.is_(True),
+        )
         if target_subject_id:
-            query = query.where(AnswerSchema.target_subject_id == target_subject_id)
+            subquery = subquery.where(AnswerSchema.target_subject_id == target_subject_id)
+
+        query: Query = select(AnswerSchema.flow_history_id, func.max(AnswerSchema.created_at))
+        query = query.where(AnswerSchema.submit_id.in_(subquery))
```

Comment:
The query looks too complex. Do we expect some answers of completed submission where answer[is_completed=false].created_at > answer[is_completed=true].created_at? If not, isn't it the same?:
```
select flow_history_id, max(created_at)
from answers
where applet_id=$1
    and flow_history_id is not null
    and is_flow_completed is true
group by flow_history_id
order by flow_history_id
```